fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricMinecartComparatorLogicRegistry.java

```diff
@@ -0,0 +1,21 @@
+package net.fabricmc.fabric.api.object.builder.v1.entity;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.vehicle.AbstractMinecartEntity;
+
+public class FabricMinecartComparatorLogicRegistry {
+	private static final Map<EntityType<?>, MinecartComparatorLogic<?>> logics = new HashMap<>();
+
+	/* @Nullable */
+	@SuppressWarnings("unchecked") // just to appease typechecker, we checked in the entity type builder that the type of the logic exactly matches the entity type
+	public static MinecartComparatorLogic<AbstractMinecartEntity> getCustomComparatorLogic(EntityType<?> type) {
+		return (MinecartComparatorLogic<AbstractMinecartEntity>) logics.get(type);
+	}
+
+	static void registerCustomLogic(EntityType<?> type, MinecartComparatorLogic<?> logic) {
```

Comment:
This could simply be named `register` like [`FabricDefaultAttributeRegistry`](https://github.com/FabricMC/fabric/blob/9354966b6c119b52108449b5c5f5622c89c5c53e/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricDefaultAttributeRegistry.java).

```suggestion
	static void register(EntityType<?> type, MinecartComparatorLogic<?> logic) {
```